Content.Server/MachineLinking/System/ReleaseSignallerSystem.cs

```diff
@@ -0,0 +1,139 @@
+        }
+    }
+
+    private void OnInit(EntityUid uid, ReleaseSignallerComponent component, ComponentInit args)
+    {
+        _signalSystem.EnsureTransmitterPorts(uid, component.Port);
+    }
+
+    private void OnDropped(EntityUid uid, ReleaseSignallerComponent component, DroppedEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (!component.Armed)
+            return;
+
+        _signalSystem.InvokePort(uid, component.Port);
+        component.Armed = false;
+        if(EntityManager.TryGetComponent<ReleaseSignallerHolderComponent>(args.User, out var holderComp))
+            holderComp.Switches.Remove(component);
+        _popupSystem.PopupEntity(Loc.GetString("release-signaller-release-self", ("device", uid)), args.User, Filter.Entities(args.User), PopupType.MediumCaution);
+        _popupSystem.PopupEntity(Loc.GetString("release-signaller-release-other", ("device", uid), ("user", args.User)), args.User, Filter.PvsExcept(args.User), PopupType.MediumCaution);
+        args.Handled = true;
+    }
+
+    private void OnUsedInHand(EntityUid uid, ReleaseSignallerComponent component, UseInHandEvent args)
+    {
+        if(args.Handled)
+            return;
+
+        args.Handled = true;
+
+        if (component.Armed == false)
+        {
+            var holder = EnsureComp<ReleaseSignallerHolderComponent>(args.User);
+            holder.Switches.Add(component);
+            component.Armed = true;
+            _popupSystem.PopupEntity(Loc.GetString("release-signaller-armed", ("device", uid)), args.User, Filter.Entities(args.User));
+        }
+        else
+        {
+            _signalSystem.InvokePort(uid, component.Port);
+            component.Armed = false;
+            _popupSystem.PopupEntity(Loc.GetString("release-signaller-release-self", ("device", uid)), args.User, Filter.Entities(args.User), PopupType.MediumCaution);
+            _popupSystem.PopupEntity(Loc.GetString("release-signaller-release-other", ("device", uid), ("person", args.User)), args.User, Filter.PvsExcept(args.User), PopupType.MediumCaution);
+        }
+    }
+
+    private void OnGetVerbs(EntityUid uid, ReleaseSignallerComponent component, GetVerbsEvent<AlternativeVerb> args)
+    {
+        if (component.Armed)
+        {
+            component.CancelToken = new CancellationTokenSource();
+            args.Verbs.Add(new AlternativeVerb()
+            {
+                Text = Loc.GetString("verb-release-signaller-disarm"),
+                Act = () =>
+                {
+                    _doAfterSystem.DoAfter(new DoAfterEventArgs(args.User, component.DisarmDelay, component.CancelToken.Token, uid)
+                    {
+                        BreakOnDamage = true,
+                        BreakOnStun = true,
+                        NeedHand = true,
+                        TargetFinishedEvent = new ReleaseSignallerDisarmCompleteEvent(args.User)
+                    });
+                }
+            });
+        }
+    }
```

Comment:
You never handle the do_after being cancelled + you should check if the canceltoken is set up front and cancel it if it already is.